assets/pages/contributions-landing/helpers/validation.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+
+// ----- Imports ----- //
+
+import type { Contrib, ContribError, Amount } from 'helpers/contributions';
+
+
+// ----- Functions ----- //
+
+function validateOtherAmount(amount: Amount, contrib: Contrib): ?ContribError {
+
+  if (amount.value === '') {
+    return 'noEntry';
+  }
+
+  const numericAmount = Number(amount.value);
```

Comment:
So, what should happen is that `input[type="number"]` returns an empty string on value lookup if a non-numeric value has been entered into the field. This gets handled by the first check for the empty string.

Example:

```
'23' -> '23'
'abc' -> ''
'345fdgdfh' -> ''
```

This does happen in almost all cases, but I've just noticed there's a really strange edge case in IE (big surprise). For some reason, in IE, although strings of non-numeric characters are correctly returned as the empty string, strings of non-numeric characters *which start with a numeric character* get returned in full.

Example:

```
'23' -> '23'
'abc' -> ''
'345fdgdfh' -> '345fdgdfh'
```

Sigh. Anyway, upshot of this is that I've added the NaN check for that edge case in IE, and in fact pulled this whole function out into a shared helper, since it's used across two pages now.